File: MusicDemoOC/MusicDemo/View/MainPageCell.m
//
//  MainPageCell.m
//  MusicDemo
//
//  Created by 沈凯 on 2017/11/7.
//  Copyright © 2017年 Ssky. All rights reserved.
//

#import "MainPageCell.h"

@interface MainPageCell ()
//图片
@property (weak, nonatomic) IBOutlet UIImageView *img;
//歌名
@property (weak, nonatomic) IBOutlet UILabel *songName;
//作者名
@property (weak, nonatomic) IBOutlet UILabel *singer;

@end

@implementation MainPageCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}
- (void)setModel:(MusicModel *)model
{
    _model = model;
    [self.img setImage:[UIImage imageNamed:model.singerIcon]];
    self.songName.text = model.name;
    self.singer.text = model.singer;
}
- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

@end

File: MusicDemoSwift/MusicDemoSwift/View/MainPageCell.swift
//
//  MainPageCell.swift
//  MusicDemoSwift
//
//  Created by 沈凯 on 2017/11/13.
//  Copyright © 2017年 Ssky. All rights reserved.
//

import UIKit

class MainPageCell: UITableViewCell {
//    图片
    @IBOutlet weak var img: UIImageView!
//    歌名
    @IBOutlet weak var songName: UILabel!
//    作者名
    @IBOutlet weak var singer: UILabel!
    
    var model: MusicModel {
        set {
            img.image = UIImage.init(named: newValue.singerIcon)
            songName.text = newValue.name
            singer.text = newValue.singer
        }
        get {
            return self.model
        }
    }
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }
    
}
